### ISSUE TITLE
Bun crashes when error occurs during fetch request when network fails

### ISSUE THREAD

**Issue by Abhinav5383** [oven-sh/bun#14530]
### How can we reproduce the crash?

I don't have a good idea how it happened, but it probably happened due to the device losing internet just after bun made a fetch request.

### Relevant log output

Bun v1.1.20 (ae194892) Linux x64
Linux Kernel v6.11.2 | glibc v2.40
CPU: sse42 popcnt avx avx2 avx512
Args: "bun" "run" "--hot" "src/index.ts"
Features: jsc Bun.stdin abort_signal(2) dotenv fetch(315) http_server spawn transpiler_cache(27) tsconfig_paths(2) tsconfig(3) 
Builtins: "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:net" "node:os" "node:path" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" 
Elapsed: 7223346ms | User: 17890ms | Sys: 1815ms
RSS: 9.22GB | Peak: 0.35GB | Commit: 9.22GB | Faults: 401

panic: Segmentation fault at address 0x0
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

error: script "dev" was terminated by signal SIGILL (Illegal instruction)
[1]    3200 illegal hardware instruction (core dumped)  bun dev

### Stack Trace (bun.report)

Bun v1.1.20 ([`ae19489`](<https://github.com/oven-sh/bun/tree/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66>)) on linux x86_64 [RunCommand]

Segmentation fault at address 0x00000000

- [`ld-temp.o:0`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/ld-temp.o#L0>): `WTF::Unicode::convertReplacingInvalidSequences`
- `??`
- [`http.zig:850`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/http.zig#L850>): `src.http.start___anon_126724`
- [`interpreter.zig:7330`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/shell/interpreter.zig#L7330>): `src.shell.interpreter.Interpreter.Builtin.Cd.handleChangeCwdErr`
- [`http.zig:2999`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/http.zig#L2999>): `src.deps.uws.NewSocketHandler`
- [`context.c:134`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/context.c#L134>): `us_socket_context_adopt_socket`
- [`bsd.c:159`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/bsd.c#L159>): `us_loop_run_bun_tick`
- [`array_hash_map.zig:2022`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/deps/zig/lib/std/array_hash_map.zig#L2022>): `src.http.HTTPThread.processEvents`
- [`string_immutable.zig:885`](<https://github.com/oven-sh/bun/blob/ae19489250ab74ee3ea7d5b50fca5d7d2b4e6d66/src/string_immutable.zig#L885>): `src.http.NewHTTPContext`

**Comment by Abhinav5383** [comment]
Nope, couldn't reproduce either on the version I was or on the latest version

**Comment by Cmion** [comment]
I'm on v1.1.42 and I experience this too on a long running hono app using `Bun.serve`

```sh
Args: "bun" "--hot" "src/index.ts"
Features: Bun.stdout abort_signal(42) fetch(672) http_server jsc transpiler_cache(20) tsconfig(12) tsconfig_paths process_dlopen(2)
Builtins: "bun:main" "detect-libc" "node:buffer" "node:crypto" "node:dns" "node:events" "node:fs" "node:fs/promises" "node:http" "node:https" "node:net" "node:os" "node:path" "node:perf_hooks" "node:process" "node:stream" "node:string_decoder" "node:timers" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "node-fetch" "node:http2"
Elapsed: 13005706ms | User: 133488ms | Sys: 40209ms
RSS: 0.17GB | Peak: 0.46GB | Commit: 0.02ZB | Faults: 7717

panic: Segmentation fault at address 0xE86E1277F587
```

**Comment by Jarred-Sumner** [comment]
This looks like it was caused by incorrect JS exception handling in our native bindings, which we've fixed in more recent versions of Bun (> v1.2.10 or later). Please let us know if you continue to run into this issue.
